src/LiveComponent/src/Test/InteractsWithLiveComponents.php

```diff
@@ -44,4 +44,24 @@ protected function createLiveComponent(string $name, array $data = [], ?KernelBr
             self::getContainer()->get('router'),
         );
     }
+
+    protected function assertComponentEmitEvent(TestLiveComponent $testLiveComponent, string $eventName, ?array $parameters = null): void
+    {
+        $eventData = $testLiveComponent->getEmittedEvent($testLiveComponent->render(), $eventName);
+
+        $this->assertNotNull($eventData, \sprintf('Event %s not emitted', $eventName));
+
+        if (null === $parameters) {
+            return;
+        }
+
+        foreach ($parameters as $key => $value) {
+            $this->assertSame($value, $eventData['data'][$key] ?? null, \sprintf('EventData (%s) is not valid', $key));
+        }
+    }
+
+    protected function assertComponentNotEmitEvent(TestLiveComponent $testLiveComponent, string $eventName): void
+    {
+        $this->assertNull($testLiveComponent->getEmittedEvent($testLiveComponent->render(), $eventName), \sprintf('Event %s emitted', $eventName));
```

Comment:
I believe it should be more helpful (if you have a better suggestion, add it! :) )
```suggestion
        $this->assertNull($testLiveComponent->getEmittedEvent($testLiveComponent->render(), $eventName), \sprintf('The component "%s" emitted the "%s" event, but it should not have.', '...', $eventName));
```